### PR TITLE
Fix Incorrect Uncompressed Error Handling in InboundMessage

### PR DESCRIPTION
**Pull request by original-brownbear**
* `CompressorFactory.compressor` does not throw any uncompressed exception on uncompressed bytes, it merely returns `null` in this case if the bytes are at least XContent so the current catch and re-throw logic is dead code
* Made it work again by throwing on a `null` return so we get a real error message instead of an NPE

### CODE REVIEW

**Review by ywelsch — CHANGES_REQUESTED** [review]
This needs tests

**Comment by original-brownbear** [comment]
@ywelsch good call, sorry about that. Adding tests showed this was still misbehaving because the uncompressed bytes wouldn't necessarily be xcontent (who knows what they would be I guess) ... I adjusted the logic a little more to not run into the safety checks for xcontent in `CompressorFactory` now and added a test.

**Review by Tim-Brooks — APPROVED** [review]
LGTM

File: server/src/main/java/org/elasticsearch/transport/InboundMessage.java [hunk 1]
```diff
@@ -121,6 +117,12 @@ InboundMessage deserialize(BytesReference reference) throws IOException {
         }
     }
 
+    static Compressor getCompressor(BytesReference message) {
```
Comment (ywelsch) [inline server/src/main/java/org/elasticsearch/transport/InboundMessage.java]: Can you add `@Nullable` for the result here?

**Review by ywelsch — APPROVED** [review]
LGTM

**Comment by original-brownbear** [comment]
Thanks @ywelsch + @tbrooks8 !

**Pull request merged**